subsys/settings/src/settings_fcb.c

```diff
@@ -249,7 +249,7 @@ static int settings_fcb_save(struct settings_store *cs, const char *name,
 	struct settings_fcb *cf = (struct settings_fcb *)cs;
 	struct fcb_entry_ctx loc;
 	int len;
-	int rc;
+	int rc = -EINVAL;
```

Comment:
I wonder: line 270 indicates that whatever the error reason has been in fcb_append, we return EINVAL.
Is this the right thing?

Reply:
In line 270 error-code can be different in order to reflect storage internal error or that storage became full.
-EINVAL is still OK IMO as it signal an error.